components/layout/fragment.rs

```diff
@@ -2730,6 +2730,14 @@ impl Fragment {
             self.style().get_box().perspective != Perspective::None
     }
 
+    /// Returns true if this fragment has a transform applied that causes it to take up no space.
+    pub fn has_non_invertible_transform(&self) -> bool {
+        let transform = self
+            .transform_matrix(&Rect::default())
+            .unwrap_or(LayoutTransform::identity());
+        !transform.is_invertible()
```

Comment:
This could potentially be: 

```rust
self.transform_matrix(&Rect::default())
    .map(|matrix| transform.is_invertible())
    .unwrap_or_else(false)
```

That would avoid creating an extra transformation matrix. Testing for invertability is sometiems expensive.